### PR TITLE
Clean up AMP logic/naming

### PR DESCRIPTION
**Pull request by ijjk**
Changes `hasAmp`, `isAmp`, `amphtml`, etc to more descriptive names to help distinguish what they are signifying. Also removes old `query` logic from dirty AMP optimizing which is no longer needed.

### CODE REVIEW

**Review by shuding — APPROVED** [review]

File: packages/next/export/worker.js [hunk 1]
```diff
@@ -130,10 +126,10 @@ process.on(
           }
         }
 
-        if (curRenderOpts.amphtml && query.amp) {
+        if (curRenderOpts.inAmpMode) {
           await validateAmp(html, path)
         }
-        if (curRenderOpts.hasAmp) {
+        if (curRenderOpts.hasAmpVersion) {
```
Comment (Timer) [inline packages/next/export/worker.js]: This logic isn't correct, right? `inAmpMode` implies the page is already AMP, so checking `hasAmpVersion` is irrelevant. This seems like legacy code (dirty amp days)... 🤔

Am I misunderstanding something?

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/export/worker.js (hunk 1 above)
Comment (ijjk) [reply packages/next/export/worker.js]: The `inAmpMode` is separate from this. This is checking if it is a hybrid page and needs to render the AMP version thus `hasAmpVersion`

Comment (Timer) [reply packages/next/export/worker.js]: Ah -- I'm still not fond of these names then:
```js
const hasAmpVersion = !inAmpMode && ampState.enabled
```

Has amp version means amp is enabled for the page, but is false when already being treated as amp? I don't like this. 🤔

Comment (ijjk) [reply packages/next/export/worker.js]: Yeah, I agree the naming is weird. Will tweak it some more

File: packages/next-server/lib/amp.ts [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
   hasQuery = false,
 } = {}) {
-  return enabled && (!hybrid || (hybrid && hasQuery))
+  return ampFirst || (hybrid && hasQuery)
 }
 
 export function useAmp() {
-  const ampMode = React.useContext(AmpModeContext)
+  const ampState = React.useContext(AmpStateContext)
   // un-comment below to not be considered AMP in dirty mode
```
Comment (Timer) [inline packages/next-server/lib/amp.ts]: This comment can go away now right? There's no such thing as dirty mode anymore.
```suggestion
```

File: packages/next-server/lib/amp.ts [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
-  return isAmp(ampMode) // && ampMode.hasQuery
+  return isInAmpMode(ampState) // && ampMode.hasQuery
```
Comment (Timer) [inline packages/next-server/lib/amp.ts]:
```suggestion
  return isInAmpMode(ampState)
```

**Review by Timer — CHANGES_REQUESTED** [review]
Also, WRT bundle size, do you think it's just because of the new context name?

**Comment by Timer** [comment]
We can address these things in a future follow up.

**Pull request merged**